common.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+option go_package = ".;pb";
+
+package common;
+
+// Identity is included in every message that requires auth. One AuthMethod at a
+// time is supported. For now, the only supported method is a token. These need
+// to be created manually and distributed to any clients.
+message Identity {
+  oneof AuthMethod {
+    string token = 1;
+  }
+}
+
+// When sent by a chat backend, the ID MAY optionally be used as a hint to the
+// frontend to get the same external ID on reconnection. When sent by core to a
+// plugin, the ID MUST be an ID pointing to an exact backend instance.
+message Backend {
+  string id = 1;
+  string type = 2;
+}
+
+// When sent by a chat backend, the ID MUST be a unique ID representing this
```

Comment:
Unique to the backend? To core? What happens if the ID isn't unique? Errors?

Reply:
Unique to core. The connection will be closed if there is already a running backend of the same type with the same ID.

Reply:
Can you write both of those explicitly here?

Reply:
Also, if things are unique to core, shouldn't core expose a "generate me a unique ID" RPC?

Reply:
Chatted offline. The IDs are meant to be a descriptive string and not a random string (what I thought you were proposing). I think calling both the structure and the behaviors on collision out explicitly in the comment is important.